presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java

```diff
@@ -3242,8 +3242,8 @@ private void testCreateTemporaryTable(
             // finish only second insert
             metadata.finishInsert(session, secondInsert, secondFragments, ImmutableList.of());
 
-            // check that there are splits for every bucket
-            assertEquals(getAllSplits(transaction, tableHandle, TupleDomain.all()).size(), bucketCount);
+            // no splits for empty buckets if zero row file is not created
+            assertLessThanOrEqual(getAllSplits(transaction, tableHandle, TupleDomain.all()).size(), bucketCount);
```

Comment:
If empty file count is known, it might be good to assert `splitCount + emptyFile = bucketCount`.

Reply:
I added a separate unit test to cover this case.